overlord/hookstate/hookmgr.go

```diff
@@ -296,16 +327,45 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 		close(hookCompleted)
 	}()
 
-	select {
-	// Hook completed; it may or may not have been successful.
-	case <-hookCompleted:
-		return buffer.Bytes(), hookError
+	var abortOrTimeoutError error
+	dyingCh := tomb.Dying()
+out:
+	for {
+		select {
+		// Hook completed; it may or may not have been successful.
```

Comment:
I realize it was like that already, but the logic would probably be more readable with these comments inside the case block. At the point where the comment is being made, we don't actually know if that's the case or not.